### PR TITLE
Created build definitions for libxml2.

### PR DESCRIPTION
**Pull request by jpakkane**

### CODE REVIEW

File: meson.build [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+    c_args : c_args,
+    include_directories : inc_dirs)
+else
+  xml2lib = static_library('xml2', sources,
+    c_args : c_args,
+    include_directories : inc_dirs)
+endif
+
+dictexe = executable('testdict', 'testdict.c',
+include_directories : inc_dirs,
```
Comment (igor-raits) [inline meson.build]: no indenting.. =(

File: meson.build [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+  'legacy.c',
+  'chvalid.c',
+  'pattern.c',
+  'xmlsave.c',
+  'xmlmodule.c',
+  'schematron.c',
+  'xzlib.c'
+]
+
+inc_dirs = include_directories('include')
```
Comment (igor-raits) [inline meson.build]: let's use `incdir` as in other projects ;)

**Comment by igor-raits** [comment]
https://github.com/mesonbuild/libxml2/settings/hooks
please add hook for our wrapdb

**Comment by igor-raits** [comment]
- [x] if new project, master has tagged commit as only commit
- [x] if new branch, it is branched from master
- [x] contains a readme.txt
- [x] contains an upstream.wrap file
- [x] download link points to authoritative upstream location
- [x] wrap repository contains only build system files
- [x] merge request is pointed to correct target branch (not master)
- [ ] wrap works
- [x] repo does not have useless top level directory (i.e. libfoobar-1.0.0)

**Comment by jpakkane** [comment]
updated and hooked

File: meson.build [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+va_list ap1, ap2;
+void func() {
+  __va_copy(ap1, ap2);
+}
+''')
+  cdata.set('HAVE___VA_COPY', 1)
+endif
+
+configure_file(input : 'config.h.meson',
+output : 'config.h',
```
Comment (igor-raits) [inline meson.build]: indent

File: meson.build [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+  endif
+endforeach
+
+cdata.set('ICONV_CONST', '')
+cdata.set('SEND_ARG2_CAST', '')
+cdata.set('GETHOSTBYNAME_ARG_CAST', '')
+
+exe_ldflags = []
+
+iconv_lib = find_library('iconv', required : false)
```
Comment (igor-raits) [inline meson.build:56]: what will happens if iconv not found?

File: config.h.meson [hunk 5]
```diff
@@ -0,0 +1,334 @@
...
+/* Have compression library */
+#mesondefine HAVE_LIBLZMA
+
+/* Define if pthread library is there (-lpthread) */
+#mesondefine HAVE_LIBPTHREAD
+
+/* Define if readline library is there (-lreadline) */
+#mesondefine HAVE_LIBREADLINE
+
+/* Have compression library */
```
Comment (igor-raits) [inline config.h.meson:100]: we don't handle this in `dependency('zlib')`

**Comment by jpakkane** [comment]
iconv is built-in to libc on platforms where it's not a library. Then there's Windows, but this won't compile on that in any case. Windows port will happen when someone does the work for it.

File: config.h.meson [hunk 6]
```diff
@@ -0,0 +1,334 @@
...
+/* Define if isnan is there */
+#mesondefine HAVE_ISNAN
+
+/* Define to 1 if you have the `isnand' function. */
+#mesondefine HAVE_ISNAND
+
+/* Define if history library is there (-lhistory) */
+#mesondefine HAVE_LIBHISTORY
+
+/* Have compression library */
```
Comment (igor-raits) [inline config.h.meson:91]: we don't handle this in `dependency('liblzma')`

**Comment by jpakkane** [comment]
Those are preprocessors and not necessary. I'd rather have this in in basic form, we can improve it later if someone really needs that.

**Comment by igor-raits** [comment]
@jpakkane okay, let's do it in the future if we need this ;)

**Comment by igor-raits** [comment]
```
[47/47] Linking target testdict
FAILED: ccache cc   -o testdict testdict.dir/testdict.c.o 'libxml2.a' 
libxml2.a(xmlschemastypes.c.o): In function `_xmlSchemaDateAdd':
/home/brain/git/mesonbuild/libxml2/libxml2-2.9.2/build/../xmlschemastypes.c:3793: undefined reference to `floor'
/home/brain/git/mesonbuild/libxml2/libxml2-2.9.2/build/../xmlschemastypes.c:3794: undefined reference to `floor'
/home/brain/git/mesonbuild/libxml2/libxml2-2.9.2/build/../xmlschemastypes.c:3811: undefined reference to `floor'
/home/brain/git/mesonbuild/libxml2/libxml2-2.9.2/build/../xmlschemastypes.c:3813: undefined reference to `floor'
/home/brain/git/mesonbuild/libxml2/libxml2-2.9.2/build/../xmlschemastypes.c:3818: undefined reference to `floor'
libxml2.a(xmlschemastypes.c.o):/home/brain/git/mesonbuild/libxml2/libxml2-2.9.2/build/../xmlschemastypes.c:3819: more undefined references to `floor' follow
libxml2.a(xmlschemastypes.c.o): In function `xmlSchemaGetCanonValue':
/home/brain/git/mesonbuild/libxml2/libxml2-2.9.2/build/../xmlschemastypes.c:5866: undefined reference to `trunc'
/home/brain/git/mesonbuild/libxml2/libxml2-2.9.2/build/../xmlschemastypes.c:5869: undefined reference to `floor'
/home/brain/git/mesonbuild/libxml2/libxml2-2.9.2/build/../xmlschemastypes.c:5872: undefined reference to `floor'
libxml2.a(xpath.c.o): In function `trio_fpclassify_and_signbit':
/home/brain/git/mesonbuild/libxml2/libxml2-2.9.2/build/../trionan.c:570: undefined reference to `__fpclassify'
libxml2.a(xpath.c.o): In function `xmlXPathFormatNumber':
/home/brain/git/mesonbuild/libxml2/libxml2-2.9.2/build/../xpath.c:3167: undefined reference to `log10'
libxml2.a(xpath.c.o): In function `xmlXPathModValues':
/home/brain/git/mesonbuild/libxml2/libxml2-2.9.2/build/../xpath.c:7599: undefined reference to `fmod'
libxml2.a(xpath.c.o): In function `xmlXPathFloorFunction':
/home/brain/git/mesonbuild/libxml2/libxml2-2.9.2/build/../xpath.c:9667: undefined reference to `fmod'
libxml2.a(xpath.c.o): In function `xmlXPathCeilingFunction':
/home/brain/git/mesonbuild/libxml2/libxml2-2.9.2/build/../xpath.c:9697: undefined reference to `fmod'
libxml2.a(xpath.c.o): In function `xmlXPathRoundFunction':
/home/brain/git/mesonbuild/libxml2/libxml2-2.9.2/build/../xpath.c:9737: undefined reference to `fmod'
libxml2.a(xpath.c.o): In function `xmlXPathStringEvalNumber':
/home/brain/git/mesonbuild/libxml2/libxml2-2.9.2/build/../xpath.c:10170: undefined reference to `pow'
libxml2.a(xpath.c.o): In function `xmlXPathCompNumber':
/home/brain/git/mesonbuild/libxml2/libxml2-2.9.2/build/../xpath.c:10256: undefined reference to `pow'
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```

@jpakkane we need to link with math library `-lm`

File: meson.build [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+  if cc.has_header(h.get(0))
+    cdata.set(h.get(1), 1)
+  endif
+endforeach
+
+cdata.set('ICONV_CONST', '')
+cdata.set('SEND_ARG2_CAST', '')
+cdata.set('GETHOSTBYNAME_ARG_CAST', '')
+
+exe_ldflags = ['-lm']
```
Comment (igor-raits) [inline meson.build:54]: what will happens if we don't have math.h?

**Comment by jpakkane** [comment]
It is required by C89, so I'm inclined to not care about platforms where this is broken. If someone finds this an issue they can file a patch.

**Comment by igor-raits** [comment]
okay. merging.

**Pull request merged**